### PR TITLE
feat:cosmos provider

### PR DESCRIPTION
**Pull request by ganchoradkov**
# Description
Added `cosmos` support to the `universal-provider` where dapps can requests such as `cosmos_signDirect, cosmos_signAmino` to a wallet peer

## How Has This Been Tested?
new integration tests
dogfooding in cosmos example dapp

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: providers/universal-provider/src/providers/solana.ts [hunk 1]
```diff
@@ -26,6 +32,25 @@ class SolanaProvider implements IProvider {
     this.namespace = Object.assign(this.namespace, namespace);
   }
 
+  public requestAccounts(): string[] {
```
Comment (bkrem) [inline providers/universal-provider/src/providers/solana.ts:35]: Intentional that the Solana provider is being updated in this branch/PR too?

**Review by bkrem — APPROVED** [review]
Clean ✨ 

Tagged @0xAsimetriq too in case he wants to take a look

**Review by xzilja — APPROVED** [review]
Nice! As we are now venturing into multi-chain support via single provider package, should we move things like `cosmos-wallet` (and sollana / ethereum) equivalents into `peerDependencies` and mark them as optional, as to not pre-bundle everything and make universal-provider leaner?

Therse peer dependencies should be mentioned in the docs as additional installs.

File: providers/universal-provider/src/providers/solana.ts (hunk 1 above)
Comment (ganchoradkov) [reply providers/universal-provider/src/providers/solana.ts:35]: It's intentional as I've refactored the `requestAccounts` fx. Before, it was requesting accounts by sending a request with `eth_getAccounts` which doesn't work outside of evm

**Comment by ganchoradkov** [comment]
>Nice! As we are now venturing into multi-chain support via single provider package, should we move things like cosmos-wallet (and sollana / ethereum) equivalents into peerDependencies and mark them as optional, as to not pre-bundle everything and make universal-provider leaner?

@0xAsimetriq the `cosmos-wallet` is just dev dependency to help sign/verify the tests. Shouldn't have an impact over prod builds

**Pull request merged**